include/boost/random/detail/uniform_int_float.hpp

```diff
@@ -60,7 +60,7 @@ class uniform_int_float
     result_type operator()()
     {
         base_result range = static_cast<base_result>((max)())+1;
-        return static_cast<result_type>(_rng() * range);
+        return static_cast<result_type>(static_cast<base_result>(_rng() * range));
```

Comment:
Fixed upstream in multiprecision and reverted this change: unfortunately adding explicit conversion operators to the expression templates caused new issues in chi_squared_distribution.hpp which required the "obvious" casts.  You win some you lose some I guess... whatever adding the explicit conversion operators to the expression templates is probably the right thing to do anyway.

I think that's all the points you raised addressed now?